### PR TITLE
Various small fixes and web-client features

### PR DESCRIPTION
**Pull request by sisou**
## What's in this pull request?

Various fixes for small issues I found during work on #<next PR>, plus two small features for the web-client.
See commits for a list.

## Pull request checklist

- [x] All tests pass. The project builds and runs.
- [x] I have resolved any merge conflicts.
- [x] I have resolved all `clippy` and `rustfmt` warnings.

### CODE REVIEW

File: web-client/src/primitives/hash.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+use nimiq_serde::Serialize;
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+pub struct Hash;
+
+#[wasm_bindgen]
+impl Hash {
+    #[wasm_bindgen(js_name = computeBlake2b)]
+    pub fn compute_blake2b(data: &[u8]) -> Vec<u8> {
```
Comment (jsdanielh) [inline web-client/src/primitives/hash.rs:11]: Could you add some small `rustdoc`?

File: web-client/src/primitives/hash.rs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        nimiq_hash::Hasher::digest(nimiq_hash::Blake2bHasher::default(), data).serialize_to_vec()
+    }
+
+    #[wasm_bindgen(js_name = computeArgon2d)]
+    pub fn compute_argon2d(data: &[u8]) -> Vec<u8> {
```
Comment (jsdanielh) [inline web-client/src/primitives/hash.rs:17]: Could you add some small `rustdoc`?

File: web-client/src/primitives/hash.rs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        nimiq_hash::Hasher::digest(nimiq_hash::Argon2dHasher::default(), data).serialize_to_vec()
+    }
+
+    #[wasm_bindgen(js_name = computeSha256)]
+    pub fn compute_sha256(data: &[u8]) -> Vec<u8> {
```
Comment (jsdanielh) [inline web-client/src/primitives/hash.rs:23]: Could you add some small `rustdoc`?

File: web-client/src/primitives/hash.rs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        nimiq_hash::Hasher::digest(nimiq_hash::Sha256Hasher::default(), data).serialize_to_vec()
+    }
+
+    #[wasm_bindgen(js_name = computeSha512)]
+    pub fn compute_sha512(data: &[u8]) -> Vec<u8> {
```
Comment (jsdanielh) [inline web-client/src/primitives/hash.rs:29]: Could you add some small `rustdoc`?

Comment (sisou) [reply web-client/src/primitives/hash.rs:29]: Haha, I'm tempted to just put `/// small` now everywhere 😅

I don't have my laptop/SSH key now, but I'll formulate the rustdoc here and then you can add it, ok?

File: web-client/src/primitives/hash.rs (hunk 1 above)
Comment (sisou) [reply web-client/src/primitives/hash.rs:11]: Do we really need a rustdoc? The method name is pretty self-explanatory?

Comment (sisou) [reply web-client/src/primitives/hash.rs:11]: > Computes a 32-byte Blake2b hash from the input data. Blake2b is used for example to compute a public key's address.

File: web-client/src/primitives/hash.rs (hunk 2 above)
Comment (sisou) [reply web-client/src/primitives/hash.rs:17]: > Computes a 32-byte Argon2 hash from the input data. Argon2d hashes are used for example for [purpose].

Please find out for what, @jsdanielh 😅

File: web-client/src/primitives/hash.rs (hunk 3 above)
Comment (sisou) [reply web-client/src/primitives/hash.rs:23]: > Computes a 32-byte SHA256 hash from the input data. SHA256 is a common hash algorithm.

File: web-client/src/primitives/hash.rs (hunk 4 above)
Comment (sisou) [reply web-client/src/primitives/hash.rs:29]: > Computes a 64-byte SHA512 hash from the input data.

Comment (jsdanielh) [reply web-client/src/primitives/hash.rs:29]: > I don't have my laptop/SSH key now, but I'll formulate the rustdoc here and then you can add it, ok?

Yup, no problem

File: web-client/src/primitives/hash.rs (hunk 2 above)
Comment (jsdanielh) [reply web-client/src/primitives/hash.rs:17]: We actually don't have a specific usage for it :smile:

**Review by jsdanielh — APPROVED** [review]

File: web-client/src/primitives/hash.rs (hunk 2 above)
Comment (sisou) [reply web-client/src/primitives/hash.rs:17]: I think it was the PoW hash.

Comment (jsdanielh) [reply web-client/src/primitives/hash.rs:17]: You are right

**Pull request merged**